### DISCUSSION TITLE
Problem to Compile QEMU

### DISCUSSION THREAD

**Discussion by daerlnaxe** [msys2/MINGW-packages#11140] [Q&A]
Hi everybody, i wrote so much  until now to describe my problem that i don't know how to begin. I need to learn to compile to use patches etc... i know there is packages.

I tried several times to follow the wiki page of QEMU and to run exe by the windows console, explorer.
There were always a problem about dll... I tried on the gitlab of QEMU, but event if Thomas Huth tried to help me he is not familiar to compiling for windows (He said it, but it's great he tried to help me).  Despesrate i asked to Christopher Reiter, and he recommended me to go here (thanks to him).

I use MSYS2 mingw x64 for all the process... and i'm not at all familiar with it, even if i program as amateur, i use Visual Studio the most of time for Windows.

Since this morning 1(am), i found my problem could be resolved by using the dll from the mingw64 directory: "\MSYS2\mingw64\" at the begining i copied one by one dll but still to have other problems (one say there were a problem with __printf__ then i asked myself how it could be possible because it's look like a native instruction). the i moved my exe file, and it works perfectly.
I think there must be a problem during the compilation.

My last lines are:

```
make[1]: Leaving directory '/home/daerlnaxe/qemu/build'
changing dir to build for make ""...
make[1]: Entering directory '/home/daerlnaxe/qemu/build'
GIT     ui/keycodemapdb meson tests/fp/berkeley-testfloat-3 tests/fp/berkeley-softfloat-3 dtc capstone slirp
[1/108] Generating qemu-version.h with a custom command (wrapped by meson to capture output)
[2/35] Generating QAPI test (include) with a custom command
make[1]: Leaving directory '/home/daerlnaxe/qemu/build'

```

I ask myself if there is not a problem to have `1/108` and `2/35` ... I mean like if  something broken (an error ?). I looked at the `makefile.ninja` to see `qemu-version.h \` for the ninja targets... could it be that ? How to resolve it if the problem come from that.

I add, all compilation i did finished with some executable mixed with *.c *.o files. i ask myself it it's normal or  if it's not the consequence of an error during compiling that stops all. I would to understand to make a clean compilation

Best regards.

Just examples, not the last tries.
[QEMU Config.txt](https://github.com/msys2/MINGW-packages/files/8389611/QEMU.Config.txt)
[Qemu Compiled.txt](https://github.com/msys2/MINGW-packages/files/8389612/Qemu.Compiled.txt)

**Comment by Biswa96** [comment]
Can you provide the steps that you have followed after cloning qemu repository?

Make sure you are using correct mingw environment. For example, in start menu, the blue 'M' icon is MINGW64 for x86_64. Also check if compiler and all the required dependencies are installed correctly.

There is a pull request to update qemu, here https://github.com/msys2/MINGW-packages/pull/11000. You can check the CI logs to get some idea.

**Comment by daerlnaxe** [comment]
yes I forgot it.

I use "MSYS2 mingw x64" all the time and there is only few instructions to compiling. They ask to use mingw64, perhaps it's here i make an error, i continue to use `MSYS2 mingw x64`
```
cd qemu
./configure --enable-gtk --enable-sdl
make
```

-------
I tried also by adding more packages  after had followed the advice to look at .cirrus.yml. As the subject was there was missing dll, it seems to oriente me to missing packages (i think).

```
	pacman -S mingw-w64-x86_64-python-sphinx
	pacman -S mingw-w64-x86_64-SDL2_image
	pacman -S mingw-w64-x86_64-ninja
	pacman -S mingw-w64-x86_64-jemalloc
	pacman -S mingw-w64-x86_64-libgcrypt
	pacman -S mingw-w64-x86_64-libssh
	pacman -S mingw-w64-x86_64-snappy
	pacman -S mingw-w64-x86_64-libusb
	pacman -S mingw-w64-x86_64-usbredir
	pacman -S mingw-w64-x86_64-nettle
	pacman -S mingw-w64-x86_64-cyrus-sasl
	pacman -S mingw-w64-x86_64-curl
	pacman -S mingw-w64-x86_64-gnutls
	pacman -S mingw-w64-x86_64-libnfs
```

But the more i added packages, the more there was dll missing. Then Thomas suggested to disabling the one that have problems, i did it but previous problems back. After a forsake of few ours, by trying to understand how to learn to compile  i saw to launch an exe compiled by MSYS2 with explorer i must add a MSYS2 dll at the root, this is how i have the idea to try to find if i didn't have already the dll somewhere.

-------


I hope i'm clear, i'm french and my bad english cumulates with my lack of knowledges about mingw... I joined a result of a compilation in my previous message, not the last but i tried also with the exe file of this compilation with the move-to-mingw64-dir method and it worked too. 

Do you know if it's normal for the last lines ?? 

```
make[1]: Leaving directory '/home/daerlnaxe/qemu/build'
changing dir to build for make ""...
make[1]: Entering directory '/home/daerlnaxe/qemu/build'
GIT     ui/keycodemapdb meson tests/fp/berkeley-testfloat-3 tests/fp/berkeley-softfloat-3 dtc capstone slirp
[1/108] Generating qemu-version.h with a custom command (wrapped by meson to capture output)
[2/35] Generating QAPI test (include) with a custom command
make[1]: Leaving directory '/home/daerlnaxe/qemu/build'
```

Thanks.

I remember suddenly , i don't know if it's important but i ran that just after installed MSYS2, i never used environment variables or thinks like that. 

I have also this problems when i use configure
![2022-03-29_20h28_44](https://user-images.githubusercontent.com/18460858/161076637-475ef181-6496-4098-8355-0748971da313.png)

**Reply by Biswa96** [reply]
It does not look normal. There are 5483 sources to compile in qemu instead of 35 or 108. It is not required to move DLLs from mingw folder. That folder is already added to PATH environment variable. Does `gcc --version` shows this at first line?

```
gcc.exe (Rev10, Built by MSYS2 project) 11.2.0
```

The required dependencies can be found in this file https://github.com/msys2/MINGW-packages/blob/master/mingw-w64-qemu/PKGBUILD. Download all the dependencies from `makedepends` and `depends` list. `${MINGW_PACKAGE_PREFIX}` equals to `mingw-w64-x86_64` for mingw64 environment. 

> I have also this problems when i use configure

Those are just warnings and can be ignored as of now.

**Reply by daerlnaxe** [reply]
Hi thanks for your answer,

I tried several times since your message because I was wondering if I made the right compilation (I have the 3dfx project next to it).
At first I saw meson and some packages seemed to be not installed for MSYs2 mingw x64, i installed what was missing.

At the end this few lines.. i wrote in addition the line i used to check packages in addition and state, before to compile
![2022-04-01_11h29_32](https://user-images.githubusercontent.com/18460858/161236583-253716a7-8bdd-4e46-9faf-1f776960efa2.png)
d
![2022-04-01_11h34_07](https://user-images.githubusercontent.com/18460858/161237326-e2c513e1-bef1-4d2a-8a5e-2118fc15d00f.png)

In all case i ran this exe files by msys2 and it's ok. With the explorer i still to have the same problem, i need external dll in the same folder because my build folder  has 5Go with source files uncompiled mixed with exe files but no dll.
I made a txt with the content of the build directory after the compilation (not for subfolders)

[Directory.txt](https://github.com/msys2/MINGW-packages/files/8396550/Directory.txt)


I look at version after git clone
```
#define QEMU_PKGVERSION "v7.0.0-rc2-16-gcace6c6f3a"
#define QEMU_FULL_VERSION "6.2.92 (v7.0.0-rc2-16-gcace6c6f3a)"
```

is it possible problem come from this version ?

I used this to build.
- Launch MSYS2 mingw x64
- git clone https://www.gitlab.com/qemu/qemu
- cd qemu
- ./configure --enable-gtk --enable-sdl
- make

For the warnings i noticed what you write, then i don't pay attention now.

Thanks.
